src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -13,12 +12,6 @@ internal static partial class ThrowHelper
         // If the exception source is this value, the serializer will re-throw as JsonException.
         public const string ExceptionSourceValueToRethrowAsJsonException = "System.Text.Json.Rethrowable";
 
-        [DoesNotReturn]
```

Comment:
Not sure I follow, why does this need to be extracted to a separate source file?

Reply:
Because the code path is present in the source generator implementation.